lib/pbench/agent/base.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import datetime
+import os
+import socket
+import sys
+from pathlib import Path
+
+import click
+import six
+
+
+class BaseCommand(metaclass=abc.ABCMeta):
+    """A base class used to define the command interface."""
+
+    def __init__(self, context):
+        self.context = context
+
+        if six.PY2:
+            click.secho("python3 required, either directly or through SCL", err="red")
+            sys.exit(1)
+        self.config = self.context.config
+
+        # very first thing to do is figure out which pbench we are
+        self.pbench_run = Path(os.environ.get("pbench_run", self.config.pbench_run))
+        if not self.pbench_run.exists():
+            click.secho(
+                f"[ERROR] the provided pbench run directory, {self.pbench_run}, does not exist",
+                err="red",
+            )
+            sys.exit(1)
+
+        # the pbench temporary directory is always relative to the $pbench_run
+        # directory
+        self.pbench_tmp = Path(os.environ.get("pbench_tmp", (self.pbench_run / "tmp")))
+        self.pbench_tmp.mkdir(parents=True, exist_ok=True)
+        if not self.pbench_tmp.exists():
+            click.secho(
+                f"[ERROR] unable to create TMP dir, {self.pbench_tmp}", err="red"
+            )
+            sys.exit(1)
+        self.pbench_log = Path(os.environ.get("pbench_log", self.config.pbench_log))
+        self.pbench_install_dir = Path(
```

Comment:
Shouldn't this, and the check below, have already been done when constructing a `PbenchAgentConfig()` object?